File: pyg/pinyougou-web-manager/src/main/webapp/js/service/brandService.js
// Service层
app.service("brandService", function ($http) {
    this.findAll = function () {
        return $http.get("/web-manager/brand/findAll.do");
    }
    this.search = function (currentPage,itemsPerPage,searchEntity) {
        return $http.post("/web-manager/brand/search.do?pageNum=" + currentPage + "&&pageSize=" + itemsPerPage, searchEntity);
    }
    this.findPage=function (currentPage,itemsPerPage) {
        return $http.get("/web-manager/brand/findPage.do?pageNum=" + currentPage + "&&pageSize=" + itemsPerPage);
    }
    this.add=function (entity) {
        return $http.post('/web-manager/brand/add.do', entity);
    }
    this.update=function (entity) {
        return $http.post('/web-manager/brand/update.do', entity);
    }
    this.findOne=function(id){
        return $http.get('/web-manager/brand/findOne.do?id=' + id);
    }
    this.delete=function (selectIds) {
        return $http.get("/web-manager/brand/delete.do?ids="+selectIds);
    }
})

File: pyg/pinyougou-interface/src/main/java/cn/itcast/core/service/BrandService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.good.Brand;
import entity.PageResult;

import java.util.List;

/**
 * @Author: keyuzhang
 * @Date: 2019/3/29 18:06
 * @Version 1.0
 */

public interface BrandService {
    //查询所有品牌结果集
    List<Brand> findAll();
    //查询分页对象
    PageResult findPage(Integer pageNum, Integer pageSize);
    //条件查询
    PageResult search(Integer pageNum, Integer pageSize,Brand brand);
    //添加
    void add(Brand brand);
    //查询单体
    Brand findOne(Long id);
    //更改
    void update(Brand brand);
    //删除
    void delete(Long[] ids);

}
